components/style/gecko/global_style_data.rs

```diff
@@ -26,6 +27,40 @@ pub struct GlobalStyleData {
     pub options: StyleSystemOptions,
 }
 
+/// std::thread wants its thread names to not have embedded nulls, whereas the
+/// Gecko profiler wants its thread names to be C-style strings.
+enum Name {
+    ForRayon,
+    ForProfiler,
+}
+
+fn thread_name_with_kind(index: usize, kind: Name) -> String {
+    let mut name = format!("StyleThread#{}", index);
+    match kind {
+        Name::ForProfiler => name.push('\0'),
+        _ => (),
+    }
+    name
+}
+
+fn thread_name(index: usize) -> String {
+    thread_name_with_kind(index, Name::ForRayon)
+}
+
+fn thread_startup(index: usize) {
+    let name = thread_name_with_kind(index, Name::ForProfiler);
```

Comment:
Rather than having the `thread_name` split, I would probably prefer just using `std::ffi::CString`.

You can do:
```rust
let name = thread_name(index);
let name = CString::new(name).unwrap();
unsafe {
    Gecko_RegisterProfilerThread(name.as_ptr());
}
```

This would make it much simpler than your current approach.

Note that `CString::new` scans the string to ensure that it doesn't contain any null char inside. Given that you generate the string yourself, and it is guaranteed to not contain any null, you can also do
```rust
let name = unsafe { CString::from_vec_unchecked(name.into()) };
```
which might be marginally faster.